base/deprecated.jl

```diff
@@ -440,3 +440,11 @@ export float32_isvalid, float64_isvalid
 @deprecate (&)(x::Char, y::Char)  Char(UInt32(x) & UInt32(y))
 @deprecate (|)(x::Char, y::Char)  Char(UInt32(x) | UInt32(y))
 @deprecate ($)(x::Char, y::Char)  Char(UInt32(x) $ UInt32(y))
+
+# 11241
+
+@deprecate is_valid_char(ch::Union(Unsigned, Integer, Char)) isvalid(Char, ch)
+@deprecate is_valid_ascii(str::Union(Vector{UInt8}, ByteString)) isvalid(ASCIIString, str)
```

Comment:
@stevengj I still don't see that that is possible.  A `Vector{UInt8}` for example... could be used in either `isvalid(ASCIIString, str)` or `isvalid(UTF8String, str)`. Same issue in the example I gave above with `Char` and `Integer`.  Can you give a concrete example of how you think it could be done?  Thanks